syntax = "proto3";

package Detector;

service DetectorService {
  rpc Ping(ProtoPing) returns (Empty);
  rpc SendDetections(ProtoDetections) returns (Empty);
}

message Empty {}

message ProtoPing {
  string id = 1;
}

message ProtoDetections {
  string id = 1;
  repeated ProtoDetection detections = 2;
}

message ProtoDetection {
  string class_name = 1;
  float score = 2;
  ProtoBoundingBox bounding_box = 3; 
}

message ProtoBoundingBox {
  float top_left_x = 1;
  float top_left_y = 2;
  float width = 3;
  float height = 4;
}
